tests/run-pass/concurrency/libc_pthread_cond.rs

```diff
@@ -0,0 +1,199 @@
+// ignore-windows: No libc on Windows
+// compile-flags: -Zmiri-disable-isolation
```

Comment:
What is this testing that `condvar.rs` cannot test?

Reply:
I have reorganized the tests and moved the testable parts up to libstd. This test now only checks that timeouts are working with both monotonic and real time clocks.